receiver/mongodbatlasreceiver/receiver.go

```diff
@@ -77,47 +81,106 @@ func (s *mongodbatlasreceiver) shutdown(context.Context) error {
 	return s.client.Shutdown()
 }
 
+type ProjectWithConfig struct {
+	Project       *mongodbatlas.Project
+	ProjectConfig *ProjectConfig
+}
+
 func (s *mongodbatlasreceiver) poll(ctx context.Context, time timeconstraints) error {
-	orgs, err := s.client.Organizations(ctx)
-	if err != nil {
-		return fmt.Errorf("error retrieving organizations: %w", err)
-	}
-	for _, org := range orgs {
-		projects, err := s.client.Projects(ctx, org.ID)
+	var projectsWithConfig []ProjectWithConfig
+
+	if len(s.cfg.Projects) > 0 {
+		for _, projectCfg := range s.cfg.Projects {
+			project, err := s.client.GetProject(ctx, projectCfg.Name)
+			if err != nil {
+				s.log.Error("error retrieving project", zap.String("projectName", projectCfg.Name), zap.Error(err))
+				continue
+			}
+			projectsWithConfig = append(projectsWithConfig, ProjectWithConfig{Project: project, ProjectConfig: projectCfg})
```

Comment:
Previously we did not need such a struct because we would process each project fully before moving on to the next. There may be a case for changing that pattern, but I think it needs to be made - OR, we just call `processProject` here instead, which would be a lot simpler codewise.

Reply:
updated calling the `processProject`